### PR TITLE
Move fipmapper from subscript to fmu-tools

### PR DESCRIPTION
**Pull request by berland**

### CODE REVIEW

File: src/fmu/tools/fipmapper/fipmapper.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+        assert isinstance(self._mapdata, dict), "FipMapper needs a dictionary"
+
+        # Determine our capabilities:
+        self.has_fip2region = "fipnum2region" in self._mapdata
+        self.has_fip2zone = "fipnum2zone" in self._mapdata
+        self.has_region2fip = "region2fipnum" in self._mapdata
+        self.has_zone2fip = "zone2fipnum" in self._mapdata
+
+    def _get_explicit_mapdata(self, yamldata: Dict[str, Any]):
+        """Fetch explicit mapping configuration from a dictionary,
```
Comment (jcrivenaes) [inline src/fmu/tools/fipmapper/fipmapper.py:88]: End first sentence with period ., not comma ,

File: src/fmu/tools/fipmapper/fipmapper.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+
+class FipMapper:
+    def __init__(
+        self,
+        yamlfile: Union[str, Path] = None,
+        mapdata: Dict[str, str] = None,
+        skipstring: Union[List[str], str] = None,
+    ):
+        """FipMapper is a utility class for being able to map between
```
Comment (jcrivenaes) [inline src/fmu/tools/fipmapper/fipmapper.py:27]: Google docstrings: First line shall be a one liner summary.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

**Review by jcrivenaes — COMMENTED** [review]
Is there documentation on this?

**Comment by berland** [comment]
> Is there documentation on this?

Only what you see in the source.

File: src/fmu/tools/fipmapper/fipmapper.py (hunk 1 above)
Comment (berland) [reply src/fmu/tools/fipmapper/fipmapper.py]: Fixed.

File: src/fmu/tools/fipmapper/fipmapper.py (hunk 2 above)
Comment (berland) [reply src/fmu/tools/fipmapper/fipmapper.py]: Fixed.

**Comment by jcrivenaes** [comment]
> Is there documentation on this?
> Only what you see in the source. The user facing part of it is documented in prtvol2csv in subscript.

Would prefer to have some documentation here also, eventually that it points to the documentation in subscript

**Comment by berland** [comment]
> > Is there documentation on this?
> > Only what you see in the source. The user facing part of it is documented in prtvol2csv in subscript.
> 
> Would prefer to have some documentation here also, eventually that it points to the documentation in subscript

As RST docs? The FipMapper class is in the current state not meant for users, only for applications (pr now only prtvol2csv). As an API, inline doc should be sufficient for FipMapper?

**Comment by jcrivenaes** [comment]
OK, if it is not meant for users it is ok

**Review by jcrivenaes — APPROVED** [review]

**Pull request merged**